src/Payment.ts

```diff
@@ -317,17 +318,55 @@ export class Payment {
     )
   }
 
-  public async getTransferInformation(
+  public async getTransferInformationForTxHash(
     txHash: string,
     options: {
       decimalsOptions?: DecimalsOptions
     } = {}
+  ): Promise<TransferInformation[]> {
+    const baseUrl = utils.buildUrl(`/transfers/`, { transactionHash: txHash })
+
+    const transferInformationList = await this.provider.fetchEndpoint<
+      TransferInformationRaw[]
+    >(baseUrl)
+    const DecimalsObjectsList = await Promise.all(
+      transferInformationList.map(async transferInformation =>
+        this.currencyNetwork.getDecimals(
+          transferInformation.currencyNetwork,
+          options.decimalsOptions || {}
+        )
+      )
+    )
+    const networkDecimalsList = DecimalsObjectsList.map(
+      decimalsObject => decimalsObject.networkDecimals
+    )
+
+    return transferInformationList.map((transferInformation, index) =>
+      this.formatTransferInformationRaw(
+        transferInformation,
+        networkDecimalsList[index]
+      )
+    )
+  }
+
+  public async getTransferInformationForId(
```

Comment:
Do we really have a reason to split them up?
Why not also have a function `getTransfer(event, options)`, where we use event id if present, and if not, fall back to transaction hash?
The reason would be that this is easier to maintain without breaking.